### PR TITLE
OS-2009 - Create_encrypted_workload_with_workload_policy

### PR DESCRIPTION
**Pull request by priyankagupte15**
TC OS-2009 - Create_encrypted_workload_with_workload_policy
image

### CODE REVIEW

File: tempest/prerequisites.py [hunk 1]
```diff
@@ -1288,3 +1288,25 @@ def network_topology(self):
...
+        # create vm
+        self.vm_id = self.create_vm(vm_cleanup=False)
+        LOG.debug("Vm ID: " + str(self.vm_id))
+
+        # Attach volume to the instance
+        self.attach_volume(self.volume_id, self.vm_id, attach_cleanup=False)
+        LOG.debug("Volume attached")
+
+        # Create secret UUID
+        self.secret_uuid = self.create_secret()
```
Comment (savitha-p) [inline tempest/prerequisites.py:1308]: set the default cleanup parameter to true for all resources, as they are used in 1 test only

File: tempest/api/workloadmgr/barbican/test_barbican.py [hunk 2]
```diff
@@ -2467,3 +2476,249 @@ def test_6_barbican(self):
                     reporting.set_test_script_status(tvaultconf.FAIL)
                     reporting.add_test_script(test[0])
                     reporting.test_case_to_write()
+            # Delete workload
+            self.workload_delete(self.wid1)
+            # Delete vm
+            self.delete_vm(self.vm_id)
```
Comment (savitha-p) [inline tempest/api/workloadmgr/barbican/test_barbican.py]: These cleanup steps for vm and volume can be handled as part of prerequisites itself, as they are not used in next test.

File: tempest/api/workloadmgr/barbican/test_barbican.py [hunk 3]
```diff
@@ -2467,3 +2476,249 @@ def test_6_barbican(self):
...
+            LOG.debug("volume deleted successfully")
+
+    # Workload policy with scheduler and retention parameter
+    @test.pre_req({'type': 'barbican_workload'})
+    @decorators.attr(type='workloadmgr_cli')
+    def test_7_barbican(self):
+        reporting.add_test_script(str(__name__) + "_Create_encrypted_workload_with_workload_policy")
+        try:
+            snapshots_list = []
+            global workload_id
```
Comment (savitha-p) [inline tempest/api/workloadmgr/barbican/test_barbican.py]: Which workload is getting used here ?

File: tempest/api/workloadmgr/barbican/test_barbican.py [hunk 4]
```diff
@@ -2467,3 +2476,249 @@ def test_6_barbican(self):
...
+            for i in range(0, len(snapshot_list_of_workload)):
+                self.snapshot_delete(
+                    self.workload_id, snapshot_list_of_workload[i])
+
+            # Delete workload
+            self.workload_delete(self.workload_id)
+            time.sleep(15)
+
+            # Delete policy
+            self.workload_policy_delete(policy_id)
```
Comment (savitha-p) [inline tempest/api/workloadmgr/barbican/test_barbican.py]: This cleanup can be handled in policy creation step itself. Any specific reason for explicitly adding the delete call?

File: tempest/api/workloadmgr/base.py [hunk 5]
```diff
@@ -2634,7 +2636,7 @@ def workload_policy_delete(self, policy_id):
             LOG.debug('WorkloadPolicyDeleted: %s' % policy_id)
             return True
         except Exception as e:
-            LOG.error("Exception in workload_policy_delete: " + str(e))
+            LOG.debug("Exception in workload_policy_delete: " + str(e))
```
Comment (savitha-p) [inline tempest/api/workloadmgr/base.py]: pls log the exception with LOG.error

File: tempest/api/workloadmgr/barbican/test_barbican.py (hunk 3 above)
Comment (priyankagupte15) [reply tempest/api/workloadmgr/barbican/test_barbican.py]: encrypted workload is created after policy creation and that is used

File: tempest/api/workloadmgr/base.py (hunk 5 above)
Comment (priyankagupte15) [reply tempest/api/workloadmgr/base.py]: done

File: tempest/api/workloadmgr/barbican/test_barbican.py (hunk 3 above)
Comment (priyankagupte15) [reply tempest/api/workloadmgr/barbican/test_barbican.py]: removed the variable

**Review by nasharatm — CHANGES_REQUESTED** [review]

**Review by nasharatm — APPROVED** [review]

**Review by savitha-p — APPROVED** [review]

**Pull request merged**